main.cpp

```diff
@@ -266,6 +266,49 @@ void MQTTGateway::mqtt_connect(mosquitto* mosq, void* obj, int rc)
     mosquitto_subscribe(mosq, NULL, "xcomfort/+/set/+", 0);
 }
 
+void MQTTGateway::mqtt_disconnect(mosquitto* mosq, void* obj, int rc)
+{
+    printf("MQTT, Disconnect (%d)\n", rc);
+    //syslog(LOG_INFO, "MQTT, Disconnect (%d)\n", rc);
+
+    if (rc)
+    {
+        sleep(1);
```

Comment:
Simple way to minimize impact of possible disconnect/connect race, we're in trouble now anyways ... but I know :)